### ISSUE TITLE
fails with python 3.7 and ubuntu???

### ISSUE THREAD

**Issue by hmaarrfk** [vpelletier/python-libaio#8]
```
$ python -m libaio.test                                                                                 
EE
======================================================================
ERROR: testBasicFunctionality (__main__.LibAIOTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mark2/git/python-libaio/build/lib/libaio/test.py", line 93, in testBasicFunctionality
    io_context.submit([fsync_block])
  File "/home/mark2/git/python-libaio/build/lib/libaio/__init__.py", line 443, in submit
    for x in block_list
  File "/home/mark2/git/python-libaio/build/lib/libaio/libaio.py", line 157, in _raise_on_negative
    raise OSError(-result, func.__name__)
OSError: [Errno 22] io_submit

======================================================================
ERROR: testPoll (__main__.LibAIOTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mark2/git/python-libaio/build/lib/libaio/test.py", line 137, in testPoll
    io_context.submit([poll_block])
  File "/home/mark2/git/python-libaio/build/lib/libaio/__init__.py", line 443, in submit
    for x in block_list
  File "/home/mark2/git/python-libaio/build/lib/libaio/libaio.py", line 157, in _raise_on_negative
    raise OSError(-result, func.__name__)
OSError: [Errno 22] io_submit

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mark2/git/python-libaio/build/lib/libaio/test.py", line 141, in testPoll
    raise SkipTest('POLL kernel support missing')
NameError: name 'SkipTest' is not defined

----------------------------------------------------------------------
Ran 2 tests in 0.104s

```

```
$ python                                                                                                
Python 3.7.3 | packaged by conda-forge | (default, Mar 27 2019, 23:01:00) 
[GCC 7.3.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import libaio
>>> libaio.__version__
```

```
(owl) mark2@xps ~/g/python-libaio master↑·8|⚑ 1                                                         
$ git log -n 1                                                                                          
commit e2732ba1edaab75fac4576f8b79638e30321f9f4 (HEAD -> master, upstream/master)
Author: Vincent Pelletier <plr.vincent@gmail.com>
Date:   Thu May 30 22:24:15 2019 +0000

    setup: Bump to 0.6 .
```

should probably add a `__version__` string to the module too.

**Comment by hmaarrfk** [comment]
Ubuntu 18.04
Linux xps 4.15.0-50-generic #54-Ubuntu SMP Mon May 6 18:46:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by vpelletier** [comment]
`git blame fs/aio.c` on current Torvalds' master looks like:
```
88a6f18b950e2 (Jens Axboe                 2018-11-24 14:46:14 -0700 1812)       switch (iocb->aio_lio_opcode) {
89319d31d2d09 (Christoph Hellwig          2016-10-30 11:42:03 -0500 1813)       case IOCB_CMD_PREAD:
7316b49c2a117 (Al Viro                    2019-03-06 18:24:51 -0500 1814)               return aio_read(&req->rw, iocb, false, compat);
89319d31d2d09 (Christoph Hellwig          2016-10-30 11:42:03 -0500 1815)       case IOCB_CMD_PWRITE:
7316b49c2a117 (Al Viro                    2019-03-06 18:24:51 -0500 1816)               return aio_write(&req->rw, iocb, false, compat);
89319d31d2d09 (Christoph Hellwig          2016-10-30 11:42:03 -0500 1817)       case IOCB_CMD_PREADV:
7316b49c2a117 (Al Viro                    2019-03-06 18:24:51 -0500 1818)               return aio_read(&req->rw, iocb, true, compat);
89319d31d2d09 (Christoph Hellwig          2016-10-30 11:42:03 -0500 1819)       case IOCB_CMD_PWRITEV:
7316b49c2a117 (Al Viro                    2019-03-06 18:24:51 -0500 1820)               return aio_write(&req->rw, iocb, true, compat);
a3c0d439e4d92 (Christoph Hellwig          2018-03-27 19:18:57 +0200 1821)       case IOCB_CMD_FSYNC:
7316b49c2a117 (Al Viro                    2019-03-06 18:24:51 -0500 1822)               return aio_fsync(&req->fsync, iocb, false);
a3c0d439e4d92 (Christoph Hellwig          2018-03-27 19:18:57 +0200 1823)       case IOCB_CMD_FDSYNC:
7316b49c2a117 (Al Viro                    2019-03-06 18:24:51 -0500 1824)               return aio_fsync(&req->fsync, iocb, true);
bfe4037e722ec (Christoph Hellwig          2018-07-16 09:08:20 +0200 1825)       case IOCB_CMD_POLL:
7316b49c2a117 (Al Viro                    2019-03-06 18:24:51 -0500 1826)               return aio_poll(req, iocb);
```

IOW, `READ{,V}` and `WRITE{,V}` are from at least 2016, but `F{,D}SYNC` (which is where `testBasicFunctionality` fails, according to the line numbers) is from march 2018 and `POLL` from mid-2018. So I am not surprised that 18.04 does not support POLL, and just a bit surprised that it also does not support `F{,D}SYNC`. Probably they pinned a kernel well before the release date, which makes it more plausible.

So I need to split the test a bit more.

**Comment by hmaarrfk** [comment]
thanks for investigating!

**Comment by vpelletier** [comment]
Should be fixed by 693a8d6314e7cfc30f0ca4a042122ca581967f79 .
